lib/_http_client.js

```diff
@@ -858,9 +858,13 @@ function listenSocketTimeout(req) {
 }
 
 ClientRequest.prototype.onSocket = function onSocket(socket, err) {
-  // TODO(ronag): Between here and onSocketNT the socket
-  // has no 'error' handler.
-  process.nextTick(onSocketNT, this, socket, err);
+  if (err) {
+    this.emit('error', err);
```

Comment:
I think this should be emitted using `process.nextTick()` like it is done here https://github.com/nodejs/node/blob/d150316a8ecad1a9c20615ae62fcaf4f8d060dcc/lib/_http_client.js#L345 otherwise the user does not have the time to add an `'error'` listener if the `agent.createConnection()` callback is called synchronously.

Reply:
@lpinca I've applied your suggestion. Thank you very much 
I will now add a test for this